flake: `TestAPI/NoUpdaterLoopLogspam`

This is a different flake to https://github.com/coder/internal/issues/766

```
--- FAIL: TestAPI (0.00s)
    --- FAIL: TestAPI/NoUpdaterLoopLogspam (0.00s)
        mock.go:448: Mock Clock - Trap TickerFunc(..., [updaterLoop])
        t.go:106: 2025-07-16 19:55:49.112 [debu]  updater loop started
        t.go:106: 2025-07-16 19:55:49.112 [debu]  watcher loop started
        t.go:106: 2025-07-16 19:55:49.112 [debu]  cleaning up subagents
        t.go:106: 2025-07-16 19:55:49.112 [debu]  cleanup subagents complete
        t.go:106: 2025-07-16 19:55:49.112 [debu]  performing initial containers update
        t.go:109: 2025-07-16 19:55:49.112 [erro]  initial containers update failed ...
            error= list containers failed:
                       github.com/coder/coder/v2/agent/agentcontainers.(*API).updateContainers
                           /home/coder/coder/agent/agentcontainers/api.go:665
                     - first error:
                       github.com/coder/coder/v2/agent/agentcontainers_test.TestAPI.func1
                           /home/coder/coder/agent/agentcontainers/api_test.go:356
        mock.go:202: Mock Clock - TickerFunc(<ctx>, 10s, <fn>, [agentcontainers updaterLoop]) call, matched 1 traps
        mock.go:357: Mock Clock - AdvanceNext()
        t.go:109: 2025-07-16 19:55:49.112 [erro]  updater loop ticker failed ...
            error= list containers failed:
                       github.com/coder/coder/v2/agent/agentcontainers.(*API).updateContainers
                           /home/coder/coder/agent/agentcontainers/api.go:665
                     - first error:
                       github.com/coder/coder/v2/agent/agentcontainers_test.TestAPI.func1
                           /home/coder/coder/agent/agentcontainers/api_test.go:356
        api_test.go:390: got log: "2025-07-16 19:55:49.112 [erro]  updater loop ticker failed ...\n    error= list containers failed:\n               github.com/coder/coder/v2/agent/agentcontainers.(*API).updateContainers\n                   /home/coder/coder/agent/agentcontainers/api.go:665\n             - first error:\n               github.com/coder/coder/v2/agent/agentcontainers_test.TestAPI.func1\n                   /home/coder/coder/agent/agentcontainers/api_test.go:356\n"
        api_test.go:396: Mock Clock - AdvanceNext()
        t.go:106: 2025-07-16 19:55:49.113 [debu]  updater loop ticker skipped, update in progress
        api_test.go:401: got log: "2025-07-16 19:55:49.113 [debu]  updater loop ticker skipped, update in progress\n"
        api_test.go:402:
                Error Trace:    /home/coder/coder/agent/agentcontainers/api_test.go:402
                Error:          Should be empty, but was 2025-07-16 19:55:49.113 [debu]  updater loop ticker skipped, update in progress
                Test:           TestAPI/NoUpdaterLoopLogspam
                Messages:       same error should not be logged again
        t.go:106: 2025-07-16 19:55:49.113 [debu]  closing API
        mock.go:202: Mock Clock - TickerFunc.Wait([agentcontainers updaterLoop]) call, matched 0 traps
        t.go:106: 2025-07-16 19:55:49.113 [debu]  api context canceled
        t.go:106: 2025-07-16 19:55:49.113 [debu]  updater loop stopped
        t.go:106: 2025-07-16 19:55:49.113 [debu]  watcher loop stopped
        t.go:106: 2025-07-16 19:55:49.113 [debu]  closed API
        mock.go:477: Mock Clock - test cleanup; will no longer log clock events
FAIL
FAIL    github.com/coder/coder/v2/agent/agentcontainers 0.283s
FAIL
```

--- a/agent/agentcontainers/api.go
+++ b/agent/agentcontainers/api.go
@@ -682,8 +682,6 @@ func (api *API) updaterLoop() {
 			} else {
 				prevErr = nil
 			}
-		default:
-			api.logger.Debug(api.ctx, "updater loop ticker skipped, update in progress")
 		}
 
 		return nil // Always nil to keep the ticker going.

